azure-toolkit-libs/azure-toolkit-mysql-lib/src/main/java/com/microsoft/azure/toolkit/lib/mysql/service/MySqlFirewallRules.java

```diff
@@ -0,0 +1,106 @@
+import lombok.AllArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@AllArgsConstructor
+public class MySqlFirewallRules {
+    private static final String NAME_PREFIX_ALLOW_ACCESS_TO_LOCAL = "ClientIPAddress_";
+    private static final String NAME_ALLOW_ACCESS_TO_AZURE_SERVICES = "AllowAllWindowsAzureIps";
+    private static final String IP_ALLOW_ACCESS_TO_AZURE_SERVICES = "0.0.0.0";
+
+    private final MySqlManager manager;
+    private final MySqlEntity mySqlEntity;
+
+    public List<MySqlFirewallRule> list() {
+        return manager.firewallRules().listByServer(mySqlEntity.getResourceGroup(), this.mySqlEntity.getName()).stream()
+            .map(this::fromFirewallRule).collect(Collectors.toList());
+    }
+
+    private MySqlFirewallRule fromFirewallRule(FirewallRule firewallRule) {
+        return new MySqlFirewallRule(manager, mySqlEntity, MySqlFirewallRule.fromFirewallRule(firewallRule));
+    }
+
+    public AbstractMySqlFirewallRuleCreator create() {
+        return new MySqlFirewallRuleCreator();
+    }
+
+    public MySqlFirewallRule enableLocalMachineAccessRule(String publicIp) {
+        Preconditions.checkArgument(StringUtils.isNotBlank(publicIp),
+            "Cannot enable local machine access to mysql server due to error: cannot get public ip.");
+        String name = getAccessFromLocalRuleName();
+        MySqlFirewallRule myFirewallRule = getAzureAccessRuleByName(name);
+        if (myFirewallRule != null) {
+            if (myFirewallRule.getEntity() != null && StringUtils.equals(myFirewallRule.getEntity().getStartIpAddress(), publicIp)) {
+                return myFirewallRule;
+            }
+            return myFirewallRule.update(publicIp, publicIp).commit();
+        } else {
+            return create().withName(name)
+                .wihStartIpAddress(IP_ALLOW_ACCESS_TO_AZURE_SERVICES)
+                .withEndIpAddress(IP_ALLOW_ACCESS_TO_AZURE_SERVICES).commit();
+        }
+    }
+
+    public MySqlFirewallRule enableAzureAccessRule() {
+        MySqlFirewallRule allowAzureAccessRule = getAzureAccessRule();
+        if (allowAzureAccessRule != null) {
+            return allowAzureAccessRule;
+        }
+        return create().withName(NAME_ALLOW_ACCESS_TO_AZURE_SERVICES)
+            .wihStartIpAddress(IP_ALLOW_ACCESS_TO_AZURE_SERVICES)
+            .withEndIpAddress(IP_ALLOW_ACCESS_TO_AZURE_SERVICES).commit();
+    }
+
+    public void disableAzureAccessRule() {
+        Optional.ofNullable(getAzureAccessRule()).ifPresent(MySqlFirewallRule::delete);
+    }
+
+    public void disableLocalMachineAccessRule() {
+        Optional.ofNullable(getAzureAccessRuleByName(getAccessFromLocalRuleName())).ifPresent(MySqlFirewallRule::delete);
+    }
+
+    private MySqlFirewallRule getAzureAccessRule() {
+        return getAzureAccessRuleByName(NAME_ALLOW_ACCESS_TO_AZURE_SERVICES);
+    }
```

Comment:
how about to use "getAzureAccessRuleByName(NAME_ALLOW_ACCESS_TO_AZURE_SERVICES)" directly?

Reply:
will refactor it in later PR